client/src/components/ColorList.js

```diff
@@ -1,40 +1,70 @@
-import React, { useState } from "react";
-import axios from "axios";
+import React, { useState, useEffect } from "react";
+import axiosWithAuth from "../utils/axiosWithAuth";
 
 const initialColor = {
   color: "",
   code: { hex: "" }
 };
-
+const token = localStorage.getItem(token);
+let id;
 const ColorList = ({ colors, updateColors }) => {
   console.log(colors);
   const [editing, setEditing] = useState(false);
   const [colorToEdit, setColorToEdit] = useState(initialColor);
+  const [addColor, setAddColor] = useState(initialColor);
 
   const editColor = color => {
     setEditing(true);
     setColorToEdit(color);
   };
-
+  useEffect(() => {
+    if (colorToEdit.id != null) id = colorToEdit.id;
+  }, [editing]);
   const saveEdit = e => {
-    e.preventDefault();
     // Make a put request to save your updated color
     // think about where will you get the id from...
     // where is is saved right now?
+    axiosWithAuth()
+      .put(`/api/colors/${id}`, colorToEdit)
```

Comment:
Instead of creating a variable `id = colorToEdit.id`, why not just write our endpoint like so?
```
.put(`/api/colors/${colorToEdit.id}`, colorToEdit)
``` 
If we write our code like this, we could remove line 9 ( `let id;` ) and remove our `useEffect()` to make the code a little cleaner.